src/libstd/num/f32.rs

```diff
@@ -481,8 +486,155 @@ mod tests {
     }
 
     #[test]
-    fn test_num_f32() {
-        test_num(10f32, 2f32);
+    fn test_nan() {
+        let nan: f32 = Float::nan();
+        assert!(nan.is_nan());
+        assert!(!nan.is_infinite());
+        assert!(!nan.is_finite());
+        assert!(!nan.is_normal());
+        assert_eq!(Fp::Nan, nan.classify());
+    }
+
+    #[test]
+    fn test_infinity() {
+        let inf: f32 = Float::infinity();
+        assert!(inf.is_infinite());
+        assert!(!inf.is_finite());
+        assert!(inf.is_positive());
+        assert!(!inf.is_nan());
+        assert!(!inf.is_normal());
+        assert_eq!(Fp::Infinite, inf.classify());
+    }
+
+    #[test]
+    fn test_neg_infinity() {
+        let neg_inf: f32 = Float::neg_infinity();
+        assert!(neg_inf.is_infinite());
+        assert!(!neg_inf.is_finite());
+        assert!(neg_inf.is_negative());
+        assert!(!neg_inf.is_nan());
+        assert!(!neg_inf.is_normal());
+        assert_eq!(Fp::Infinite, neg_inf.classify());
+    }
+
+    #[test]
+    fn test_zero() {
+        let zero: f32 = Float::zero();
+        assert_eq!(0.0, zero);
+        assert!(!zero.is_infinite());
+        assert!(zero.is_finite());
+        assert!(zero.is_positive());
```

Comment:
Hm... not a problem with this patch at all, since this behaviour is documented, but I'm unsure if considering zero to be positive is a great idea. (I filed https://github.com/rust-lang/rust/issues/22985.)

Maybe you ensure that all the `f32`/`f64` tests of this nature test both `is_positive` and `is_negative`, though. I.e. check they're false when they should be, as well as checking that they're true when they should be. (Including the NaN ones.)